Ia 1696 org unit map to function

quang-le opened the pull request:
Convert OrgUnitMap to function to enable leaflet upgrade

Related JIRA tickets : IA-1696

## Self proof reading checklist

- [X] Did I use eslint and black formatters
- [X] Is my code clear enough and well documented
- [X] Are my typescript files well typed
- [ ] Did I add translations
- [ ] My migrations file are included
- [ ] Are there enough tests

## Changes

- Added basic typing to `mapUtils`
- Converted to typescript:
    - Added dedicated `types` file
    - Deleted duplicate types file for org unit types (in assignments)
    - Moved some values in `constant.ts` 
- Converted to function component
    - Replaced life-cycle functions with `useEffect`. in particular: simplified the condition triggering `fitToBounds`
    - Grouped redux code in `useRedux` hook
    - Removed calls to `fitToBounds` from filter, as it was now unnecessary
- Split up code:
    - Created dedicated components for shapes and markers
    - Left the code related to the `EditableGroup`s (location and catchment) in the main `OrgUnitMap` component as it makes heavy use of the map ref. 

- Fixed a small UI bug with `OrgUnitPopUpComponent`'s link
      

## How to test

Go to an org unit detail's map and try out all features. Also try deep linking and navigating from the Treeview and the Submissions table

## Print screen / video

Pop up button bug fix: you can click anywhere on the button and be redirected. Also: the text is not underlined on hover. For the rest nothing should change

https://user-images.githubusercontent.com/38907762/215134052-bf3cadc1-efdd-484d-aca7-8f8d0876556f.mov


## Notes

- To compare the changes in the code, this is the commit with the changes before I started to split up the code: 106e4c699dd1d7c4fd1c9fe5015f712fa31f6c21
- To test by switching between the old and new component, this is the last commit before I deleted the old js component (unfortunately I sttill had to make some changes afterwards): 40ac7f40f5cbe53a67ee9be38459e62df975fd08

olethanh commented:
the big problem I just saw is that this is not listed in the assignable permissions dialog of Iaso, so if new users are created, it won't be possible to assign the Polio permissions to them.

olethanh commented:
nevermind previous comment it was correctly listing the perm.

Updated the PR to:

1. translate the new permission
2. don't list it (from the API) if the plugin is not enabled
3. migrate existing user to the permissions
4. use the perm in the Polio Configuration menu
5. make the Permission dialog not crash it they are untranslated permissions

quang-le committed:
Merge branch 'main' into IA-1696_OrgUnitMap_to_function

quang-le committed:
IA-1696: move const and helpers in utils

quang-le committed:
Ia-1696: add utils.ts

quang-le committed:
IA-1696: move SOurceShape in own component

quang-le committed:
Ia-1696: add OrgUnitMap.tsx

quang-le committed:
IA-1696: fix catchment group

quang-le committed:
Merge branch 'main' into IA-1696_OrgUnitMap_to_function

quang-le committed:
IA-1696: remove console error

quang-le committed:
IA-1696: fixlocation and catchment

quang-le committed:
IA-1696: remove unnecessary condition

quang-le committed:
IA-1696: add typing to mapUtils

quang-le committed:
IA-1696: add typing to utils

quang-le committed:
IA-1696: update outdated import

quang-le committed:
IA-1696: add constants.ts

quang-le committed:
IA-1696: refactor getMarkersList as MarkersList

quang-le committed:
IA-1696: add new custom types

quang-le committed:
IA-1696: move redux code in dedicated hook

quang-le committed:
IA-1696: split map elements in separate components

quang-le committed:
IA-1696: light cleanup

quang-le committed:
IA-1696: delete js component

quang-le committed:
Merge remote-tracking branch 'origin' into IA-1696_OrgUnitMap_to_function

quang-le committed:
IA-1696: add typing to MarkersList

quang-le committed:
IA-1696: reduce duplicated code

- replace SourcesSelectedMarkers, OrgUnitTypesSelectedMarkers
- with new SelectedMarkers component

quang-le committed:
Ia-1696: move buttonsInitialState to constants

quang-le committed:
IA-1696: fix PopUp button

quang-le committed:
IA-1686: add explanation comment

quang-le committed:
Merge branch 'IA-1697_InstancesPopUp_to_function' into IA-1696_OrgUnitMap_to_function

beygorghor commented on hat/assets/js/apps/Iaso/domains/orgUnits/components/orgUnitMap/OrgUnitMap/OrgUnitMap.tsx:
```diff
@@ -0,0 +1,568 @@
+    const hasMarker =
+        Boolean(currentOrgUnit.latitude) && Boolean(currentOrgUnit.longitude);
+
+    if (map.current) {
+        map.current.leafletElement.options.maxZoom = currentTile.maxZoom;
+    }
```
this is one is a bit weird... i assume it will disappear while upgrading leaflet

quang-le replied:
That's what I assumed as well

beygorghor reviewed (approved):
Nice job ! i'm quiet sure the map is faster now, i have tried to edit org unit with a big shape and no lag at all !
Only one small thing, maybe not related:
![Screenshot 2023-02-01 at 16 25 35](https://user-images.githubusercontent.com/12494624/216086297-e5470476-5301-4e4c-af65-412646290b96.png)
I have the same form several times in the dropdown

quang-le committed:
IA_1696: fix duplicate forms in filter dropdown

quang-le commented:
cypress
